### ISSUE TITLE
Proposal: Favor unquoted atoms when quotes are unnecessary

### ISSUE THREAD

**Issue by blatyo** [rrrene/credo#447]
People on my team deal with a lot of JSON and end up writing the Elixir equivalent of JSON in the language. Because of that, they often make the mistake of leaving a colon after a quoted string, which is an atom in Elixir.

They have this:

``` elixir
%{"x": 1}
```

Instead of:

``` elixir
%{"x" => 1}
```

If credo complained about quotes being used for an atom when not necessary, it would be helpful for them and me to find these types of issues.

**Comment by NobbZ** [comment]
I do sometimes (quite often do be honest) write all my atoms in quotes (especially when used as keys), when there is a single one that **needs** to be quoted in the current "scope", to be consistent.

Therefore I'd like to have just a linter which is nearly the other way round.

Since here we have opinions which are pretty much the opposite of each other, we can't have both in core, therefore I'd suggest to create a collection of custom linters which could then easily be copy pasted in situations like these.

**Comment by blatyo** [comment]
I would say, which way things go is based on community preference. If anyone doesn't like a check, they have the ability to configure/disable it.
